webbpsf/webbpsf_core.py

```diff
@@ -688,51 +760,68 @@ def __init__(self, *args, **kwargs):
         self.pupil = os.path.abspath(self._datapath+"../pupil_RevV.fits")
         "Filename *or* fits.HDUList for JWST pupil mask. Usually there is no need to change this."
 
-        self.detector_list = ['Default']
         self._detector = None
 
         # where is the source on the detector, in 'Science frame' pixels?
-        self.detector_coordinates = (0, 0)
+        self.detector_position = (1024, 1024)
 
-    @property
-    def detector(self):
-        """Currently selected detector name (for instruments with multiple detectors)"""
-        return self._detector.name
+        self.include_si_wfe = False
+        """Should calculations include the Science Instrument internal WFE?"""
+
+
+    def _getOpticalSystem(self,fft_oversample=2, detector_oversample = None, fov_arcsec=2, fov_pixels=None, options=None):
+        # invoke superclass version of this
+        # then add a few display tweaks
+        optsys = SpaceTelescopeInstrument._getOpticalSystem(self,
+            fft_oversample=fft_oversample, detector_oversample=detector_oversample, fov_arcsec=fov_arcsec, fov_pixels=fov_pixels,
+            options=options)
+        optsys.planes[0].display_annotate = utils.annotate_ote_entrance_coords
+        #optsys.planes[-2].display_annotate = utils.annotate_sky_pupil_coords
+        return optsys
+
+    def _get_aberrations(self):
+        """ Compute field-dependent aberration for a given instrument
+
+        This is just a placeholder!
+        """
+        if not self.include_si_wfe:
+            return None
+
+        #tmp = poppy.zernike.opd_from_zernikes([0,0,0,5e-8, 1e-8],
+                #npix=1024, outside=0)
+        tmp = poppy.zernike.opd_from_zernikes([0,0,0], npix=1024, outside=0)
```

Comment:
OK, we can worry about caching/memoizing the ZernikeWFE calculations - at some later date!